app.rb

```diff
@@ -0,0 +1,113 @@
+require './book'
+require './rental'
+require './person'
+require './student'
+require './teacher'
+
+class App
+  attr_accessor :books, :rentals, :people, :student, :teacher
+
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def list_books
+    @books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def list_people
+    @people.each do |person|
+      puts "[#{person.class.name}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def create_person
+    person_type = person_type_input
+    if person_type == 1
+      person = create_student
+    elsif person_type == 2
+      person = create_teacher
+    else
+      puts 'Invalid input.'
+      return
+    end
+    push_person_to_list(person)
+    puts 'Person created successfully'
+  end
+
+  def person_type_input
+    puts 'Do you want to create a student (1) or a teacher (2)? [Input the number]:'
+    gets.chomp.to_i
+  end
+
+  def create_student
+    puts 'Age:'
+    age = gets.chomp
+    puts 'Name:'
+    name = gets.chomp
+    puts 'Has Parent Permission? [Y/N]:'
+    parent_permission = gets.chomp.downcase == 'y'
+    Student.new(age, name, parent_permission)
+  end
+
+  def create_teacher
+    puts 'Age:'
+    age = gets.chomp
+    puts 'Name:'
+    name = gets.chomp
+    puts 'Specialization:'
+    specialization = gets.chomp
+    Teacher.new(age, name, specialization)
+  end
+
+  def push_person_to_list(person)
+    @people.push(person)
+  end
+
+  def create_book
+    puts 'Title:'
+    title = gets.chomp
+    puts 'Author:'
+    author = gets.chomp
+    book = Book.new(title, author)
+    @books.push(book)
+    puts 'Book created successfully'
+  end
+
+  def create_rental
+    puts 'Select a book from the following list by number:'
+    @books.each_with_index do |book, index|
+      puts "#{index}) Title: #{book.title}, Author: #{book.author}"
+    end
+    book = gets.chomp.to_i
+
+    puts 'Select a person from the following list by number (not id):'
+    @people.each_with_index do |person, index|
+      puts "#{index}) [#{person.class.name}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    person = gets.chomp.to_i
+
+    puts 'Date (YYYY-MM-DD):'
+    date = Date.parse(gets.chomp)
+
+    rental = Rental.new(date, @books[book], @people[person])
+    @rentals.push(rental)
+    puts 'Rental created successfully'
+  end
+
```

Comment:
![image](https://github.com/yayaismaail/oop-school-library/assets/93039359/10a5acbb-09d6-418e-b6ba-dd03a7302ecd)

- [ ] It would be nice if you fix the issue of the Student name while creating a rental because as you can see in the image above :eyes: that instead of Student name I am getting true :fearful: Which is not according to the project requirements :thinking: Kindly fix it to make your app work as expected :+1: